crates/core/tests/client.rs

```diff
@@ -1,22 +1,137 @@
     use kakarot_rpc_core::mock::wiremock_utils::setup_mock_client_crate;
     use kakarot_rpc_core::models::block::BlockWithTxs;
     use kakarot_rpc_core::models::convertible::{ConvertibleStarknetBlock, ConvertibleStarknetEvent};
     use kakarot_rpc_core::models::event::StarknetEvent;
-    use kakarot_rpc_core::models::felt::Felt252WrapperError;
+    use kakarot_rpc_core::models::felt::{Felt252Wrapper, Felt252WrapperError};
     use kakarot_rpc_core::models::ConversionError;
     use reth_primitives::{Address, Bytes, H256};
     use reth_rpc_types::Log;
     use starknet::core::types::{BlockId, BlockTag, Event, FieldElement};
     use starknet::core::utils::get_selector_from_name;
     use starknet::providers::Provider;
 
+    use crate::helpers::constants::{EVM_ADDRESS, EVM_PRIVATE_KEY};
+    use crate::helpers::deploy_helpers::{create_raw_ethereum_tx, deploy_kakarot_system};
+
+    #[tokio::test]
+    #[ignore]
+    async fn test_constructable() {
+        // initial setup of Constructable to test we can deploy contracts w/ constructor arguments
+        let starknet_test_sequencer = TestSequencer::start().await;
+
+        let eoa_private = H256::from_slice(&hex::decode(EVM_PRIVATE_KEY).unwrap());
+
+        let expected_funded_amount = FieldElement::from_dec_str("100000000000000000").unwrap();
+
+        let deployed_kakarot =
+            deploy_kakarot_system(&starknet_test_sequencer, EVM_ADDRESS, eoa_private, expected_funded_amount).await;
+
+        let address = "0x0000000000000000000000000000000000000000".parse::<EthersAddress>().unwrap();
+        let (_constructable_abi, deployed_addresses) = deployed_kakarot
+            .deploy_evm_contract(
+                starknet_test_sequencer.url(),
+                "Constructable",
+                // more than one argument to a constructor needs to be conveyed as a tuple
+                (EthersU256::from(100), address),
+            )
+            .await
+            .unwrap();
+
+        let kakarot_client = KakarotClient::new(StarknetConfig::new(
+            starknet_test_sequencer.url().as_ref(),
+            deployed_kakarot.kakarot,
+            deployed_kakarot.kakarot_proxy,
+        ))
+        .unwrap();
+
+        let Ok(opcode_eth_addr) = ({
+            let address : Felt252Wrapper = (*deployed_addresses.first().unwrap()).into();
+            address.try_into()
+        }) else {
+            panic!("returned deploy address from kakarot failed to convert to address")
+        };
+
+        kakarot_client.get_code(opcode_eth_addr, BlockId::Tag(BlockTag::Latest)).await.expect("contract not deployed");
+    }
+
+    #[tokio::test]
+    #[ignore]
+    async fn test_counter() {
+        let starknet_test_sequencer = TestSequencer::start().await;
+
+        let eoa_public = Address::from_slice(&hex::decode(EVM_ADDRESS.trim_start_matches("0x")).unwrap());
+        let eoa_private = H256::from_slice(&hex::decode(EVM_PRIVATE_KEY).unwrap());
```

Comment:
we should strive to make things as clear as possible, even when it's seemingly obvious, because in other places, it may not be obvious for some

eoa_private -> `eoa_private_key`